File: FormalBench-data/diverse/rule_23/MaxProduct.java

import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;

class MaxProduct {

	public static int maxProduct(int[] arr, int n) {
		int[] mpis = new int[n];
		for (int i = 0; n > i; i++) {
			mpis[i] = arr[i];
		}
		for (int i = 1; n > i; i++) {
			for (int j = 0; i > j; j++) {
				if (arr[j] < arr[i] && (mpis[j] * arr[i]) > mpis[i]) {
					mpis[i] = mpis[j] * arr[i];
				}
			}
		}
		int max = mpis[0];
		for (int i = 1; n > i; i++) {
			if (max < mpis[i]) {
				max = mpis[i];
			}
		}
		return max;
	}
}

File: FormalBench-data/FilteredData/translated_c/seed_726_per_category_10_653ade686f/MaxProduct.c
#include "java_arrays.h"
#include <stdint.h>

static int32_t java_add(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left + (uint32_t)right);
}

static int32_t java_mul(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left * (uint32_t)right);
}

int32_t maxProduct(JIntArray arr, int32_t n) {
    JIntArray mpis = jarray_new(n);
    for (int32_t i = 0; i < n; i = java_add(i, 1)) {
        jarray_set(mpis, i, jarray_get(arr, i));
    }
    for (int32_t i = 1; i < n; i = java_add(i, 1)) {
        for (int32_t j = 0; j < i; j = java_add(j, 1)) {
            int32_t current = jarray_get(arr, i);
            int32_t previous = jarray_get(arr, j);
            int32_t product = java_mul(jarray_get(mpis, j), current);
            if (current > previous && jarray_get(mpis, i) < product) {
                jarray_set(mpis, i, product);
            }
        }
    }
    int32_t max = jarray_get(mpis, 0);
    for (int32_t i = 1; i < n; i = java_add(i, 1)) {
        if (jarray_get(mpis, i) > max) {
            max = jarray_get(mpis, i);
        }
    }
    return max;
}
